### ISSUE TITLE
[Rockon] When Rockon requires no environment variables, summary screen is not shown before installation

### ISSUE THREAD

**Issue by Hooverdan96** [rockstor/rockstor-core#2934]
During test installations of a submitted Rockon (and observed but ignored before), the symptom appears that if he "last" screen to configure is not the `environment` variables screen in the configuration wizard, no summary page is shown at all.

For example (for the adguard-home Rockon):
When reaching the ports screen:
<img width="640" alt="image" src="https://github.com/user-attachments/assets/b29ec897-d76b-437e-a980-0def7ee6cc8e">

and then select Next, the expected behavior would be the summary page (as no environment variables need to be defined). Instead, nothing happens. When pressing the `Next` button a second time, the installation is immediately initiated, instead of showing the summary screen:
<img width="640" alt="image" src="https://github.com/user-attachments/assets/392beac0-ebbb-4fd3-951b-7d0e90adc33f">

Once the installation is complete/successful, the summary page can still be viewed the normal way. 
<img width="583" alt="image" src="https://github.com/user-attachments/assets/2c6335ef-d719-4eb8-b58f-638133b87c5d">

Note: I don't think this is a high priority issue to be addressed, since, aside from the pending adguard-home Rockon, I don't believe we currently have any other Rockons with that configuration, but wanted to report it regardless.

**Comment by FroggyFlox** [comment]
Nice find, @Hooverdan96 !

To be honest, I'm surprised we haven't encountered something like that sooner as I've always found the way the page counter for the rock on install wizard a bit fragile. My best guess is something that has not accounted for missing "objects" (devices, ports, env,...) in the corresponding view: 
https://github.com/rockstor/rockstor-core/blob/812f1cffe39b0c7b6005215355b7fa413feae4ea/src/rockstor/storageadmin/static/storageadmin/js/views/rockons.js#L425

By curiosity, what error do you see in the browser developer console when clicking the "Next" button the first time (when nothing happens)?

**Comment by Hooverdan96** [comment]
I'll try that again and check. I'll get back to you

**Comment by Hooverdan96** [comment]
```
Uncaught TypeError: Cannot convert undefined or null to object           https://<Rockstor address>/static/storageadmin/js/storageadmin.js
    at Function.entries (<anonymous>)
    at child.render (storageadmin.js:13951:58)
    at child.renderCurrentPage (storageadmin.js:665:54)
    at child.incrementPage (storageadmin.js:643:18)
    at Object.<anonymous> (storageadmin.js:624:19)
    at c (jquery-1.9.1.min.js:3:7857)
    at Object.add [as done] (jquery-1.9.1.min.js:3:8167)
    at child.nextPage (storageadmin.js:623:17)
    at HTMLDivElement.dispatch (jquery-1.9.1.min.js:3:28337)
    at v.handle (jquery-1.9.1.min.js:3:25042)
```

**Comment by phillxnet** [comment]
@Hooverdan96 is this not a duplicate of: 

- OpenVPN rock-on install wizard fails to show pre-install summary #2913

Or at least this may have narrowed down the/a possible cause.

**Comment by Hooverdan96** [comment]
I think you are correct, since the OpenVPN has a similar setup with no environment variables surfaced for user interaction. I will close this one and we can keep the original Issue open.

**Comment by Hooverdan96** [comment]
Added comment to #2913, closing this issue.
